sklearn/feature_selection/_sequential.py

```diff
@@ -155,16 +195,22 @@ def fit(self, X, y):
         )
         n_features = X.shape[1]
 
+        # FIXME: to be fixed in 1.3
         error_msg = (
-            "n_features_to_select must be either None, an "
-            "integer in [1, n_features - 1] "
+            "n_features_to_select must be either 'auto', 'warn', "
+            "None, an integer in [1, n_features - 1] "
             "representing the absolute "
             "number of features, or a float in (0, 1] "
             "representing a percentage of features to "
             f"select. Got {self.n_features_to_select}"
         )
-        if self.n_features_to_select is None:
-            self.n_features_to_select_ = n_features // 2
+        if self.n_features_to_select in ("auto", "warn", None):
```

Comment:
In the edge case where `tol is not None` and `n_features_to_select=None`, we need to include the same code path as before:

```python
if self.n_features_to_select in ("warn", None):
    if self.tol is not None:
		raise ValueError("tol is only enabled if `n_features_to_select='auto'`")
    self.n_features_to_select_ = n_features // 2
elif self.n_features_to_select == "auto":
```

Reply:
fixed to be addressed until 1.3 https://github.com/scikit-learn/scikit-learn/pull/20145/commits/1ba4b4cbd2f5a7d0556114d06e734821ff70080c

fixed to include the same code path https://github.com/scikit-learn/scikit-learn/pull/20145/commits/23f722c686af450b8664ae53cdf59a7fed76d396